File: software_challenge_sdk/src/server_api/sc/protocol/responses/CloseConnection.java
package sc.protocol.responses;

import com.thoughtworks.xstream.annotations.XStreamAlias;

/**
 * Is sent by one party immediately before this party closes the communication
 * connection and should make the receiving party also close the connection.
 */
@XStreamAlias(value="close")
public class CloseConnection extends ProtocolMessage
{

}

File: sdk/src/server-api/sc/protocol/CloseConnection.kt
package sc.protocol

import com.thoughtworks.xstream.annotations.XStreamAlias

/**
 * Is sent by one party immediately before this party closes the communication
 * connection and should make the receiving party also close the connection.
 *
 * This should not be sent manually, the XStreamClient will automatically send
 * it when stopped.
 */
@XStreamAlias(value = "close")
class CloseConnection: ProtocolPacket {
    override fun toString(): String = javaClass.simpleName
}
